src/Psl/IO/Reader.php

```diff
@@ -0,0 +1,142 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Psl\IO;
+
+use Psl;
+use Psl\Str;
+
+final class Reader implements ReadHandle
```

Comment:
Are you also planning to add a writer?

Reply:
I think not, if we would have a writer, it would probably only contain `writeLine` beside what already exists in `WriteHandle`, and that is pretty simple to do yourself ( write($text."\n"); )
unless you have other things in mind that we can add in a writer?

Reply:
Some other possible additions to a writer that could make it all easier to use:

- `writeLine`
- `copy(SeekableReadHandle)` as-in `stream_copy_to_stream()`.
- `writeALot(iterable)`
- ???

For Stdout you could add some terminal specific things, but that should most likely go in it's own specific writer.

We can always add this stuff if we need it somewhere!

Reply:
> copy(SeekableReadHandle)

i think that can be a function in its own `IO\copy(ReadHandle $source, WriteHandle $destination, ?int $chunk_size = null, ?float $timeout_seconds = null)`

> writeALot(iterable)

this is a one liner :) 
`Iter\apply($iterable, fn(string $content) => $handle->write($content));`

> terminal specific things

we don't really want that, we want `IO` to contain general purpose IO things, as the same handles will be used for files and network sockets.

Reply:
added writer :)